runtime/src/bank.rs

```diff
@@ -1104,7 +1104,31 @@ impl Bank {
         )
     }
 
-    pub fn new_with_paths(
+    pub fn new_with_paths_for_tests(
+        genesis_config: &GenesisConfig,
+        paths: Vec<PathBuf>,
+        frozen_account_pubkeys: &[Pubkey],
+        debug_keys: Option<Arc<HashSet<Pubkey>>>,
+        additional_builtins: Option<&Builtins>,
+        account_indexes: AccountSecondaryIndexes,
+        accounts_db_caching_enabled: bool,
+        shrink_ratio: AccountShrinkThreshold,
+        debug_do_not_add_builtins: bool,
+    ) -> Self {
+        Self::new_with_paths_production(
+            genesis_config,
+            paths,
+            frozen_account_pubkeys,
+            debug_keys,
+            additional_builtins,
+            account_indexes,
+            accounts_db_caching_enabled,
+            shrink_ratio,
+            debug_do_not_add_builtins,
+        )
+    }
+
+    pub fn new_with_paths_production(
```

Comment:
In all the previous PRs the functions without `_for_tests` meant for production. Why the switch here? (I am assuming the "production" here means not tests/benches.)

Reply:
Yes, that function is about to be refactored away. I did not want to leave the old, non-test function around and allow other callers to call it in the meantime accidentally. It is confusing. Trying to take clear, obvious steps towards cleaving the paths into thirds: benches, tests, production/normal/non-test-or-bench